run/rails-cat_album/app/helpers/photos_helper.rb

```diff
@@ -1,2 +1,11 @@
 module PhotosHelper
+  def image_url photo
+    case Rails.application.config.active_storage.service
+    when :local
+      url = url_for photo.image
+    when :google
+      url = "https://storage.googleapis.com/cat_album_storage/#{photo.image.key}"
+    end
+    url
```

Comment:
Some Ruby tips:

Every Ruby construct has a "return value". In this case, you have a `case` statement, which is the final value of whatever "when" clause gets selected. In other words, what you have above:

```ruby
case Rails.application.config.active_storage.service
when :local
  url = url_for photo.image
when :google
  url = "https://storage.googleapis.com/cat_album_storage/#{photo.image.key}"
end
```

is equivalent to:

```ruby
url = case Rails.application.config.active_storage.service
      when :local
        url_for photo.image
      when :google
        "https://storage.googleapis.com/cat_album_storage/#{photo.image.key}"
      end
```

If you make that change, your method as a whole would look like this:

```ruby
def image_url photo
  url = case Rails.application.config.active_storage.service
        when :local
          url_for photo.image
        when :google
          "https://storage.googleapis.com/cat_album_storage/#{photo.image.key}"
        end
  url
end
```

At that point, you don't need the `url` variable at all anymore because the return value of the method is the return value of the last statement in the method, which is the `case` statement. So this is now equivalent to:

```ruby
def image_url photo
  case Rails.application.config.active_storage.service
  when :local
    url_for photo.image
  when :google
    "https://storage.googleapis.com/cat_album_storage/#{photo.image.key}"
  end
end
```

Reply:
Ahh I see, this makes a lot of sense and looks more clean. TY for the tip Daniel!